org.scala-ide.sdt.core/src/org/scalaide/ui/internal/editor/ScalaSourceFileEditor.scala

```diff
@@ -439,7 +440,35 @@ class ScalaSourceFileEditor
       else
         super.requestWidgetToken(requester, priority)
   }
+  private var outlinePage:ScalaOutlinePage = null
```

Comment:
Overall, please try to move all of this outline stuff to a trait called `OutlineEditorExtension`, similar to how it is done in `MarkOccurrencesEditorExtension` and mix it into the `ScalaCompilationUnitEditor`.